.circleci/config.yml

```diff
@@ -1,4 +1,54 @@
-version: 2
+version: 2.1
+
+commands:
+  m2-php-test:
+    description: PHP Unit tests for M2
+    parameters:
+      cache_key_separator:
+        type: string
+      m2_version:
+        type: enum
+        default: "2.3.0"
```

Comment:
Perhaps we should always use the latest version of Magento (currently it is 2.3.3) as well as the oldest we support (I think 2.2.0 is OK, although we have some merchants on 2.1.x)

Reply:
that sounds like a reasonable push. I would like to however not make that a part of this PR since this is a refactor. I will start a chat on slack tomorrow to see why we havent started testing against 2.3.3 yet